### PR TITLE
#2559. Add more augmenting expressions tests

### PR DESCRIPTION
**Pull request by sgrekhov**

### CODE REVIEW

**Review by eernstg — COMMENTED** [review]
Looks good, except that a couple of descriptions seems to need an update.

File: LanguageFeatures/Augmentation-libraries/augmented_expression_A06_t03.dart [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+/// @assertion The exact result of an `augmented` expression depends on what is
+/// being augmented, but it generally follows the same rules as any normal
+/// identifier:
+/// ...
+/// - Augmenting enum values: When augmenting an enum value, augmented has no
+///   meaning and is not allowed.
+///
+/// @description Checks that it is a compile-time error to augment an enum value
+/// with has a constructor with a parameter named `augmented`.
```
[the remark this answers is not in the document]
Comment (sgrekhov) [reply LanguageFeatures/Augmentation-libraries/augmented_expression_A06_t03.dart]: Thank you! Updated.

File: LanguageFeatures/Augmentation-libraries/augmented_expression_A06_t04.dart [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+/// @assertion The exact result of an `augmented` expression depends on what is
+/// being augmented, but it generally follows the same rules as any normal
+/// identifier:
+/// ...
+/// - Augmenting enum values: When augmenting an enum value, augmented has no
+///   meaning and is not allowed.
+///
+/// @description Checks that it is a compile-time error to augment an enum value
+/// with a parameter whose name is `augmented`.
```
[the remark this answers is not in the document]
Comment (sgrekhov) [reply LanguageFeatures/Augmentation-libraries/augmented_expression_A06_t04.dart:13]: But in this test we are augmenting not an enum value named `augmented` but an enum value with parameter named `augmented`. PTAL

**Review by sgrekhov — COMMENTED** [review]
Partially updated.  Please take another lookk at description of `augmented_expression_A06_t04.dart`

File: LanguageFeatures/Augmentation-libraries/augmented_expression_A06_t04.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (eernstg) [reply LanguageFeatures/Augmentation-libraries/augmented_expression_A06_t04.dart:13]: Ah, yes!

The confusion perhaps came from the use of the word 'parameter'. The specification documents are pretty much consistent (and exceptions should be fixed! ;-) in using 'parameter' and 'formal parameter' to denote the declaration of a formal parameter of a function, and using 'argument' and 'actual argument' for the expression that provides a value which will be bound to the corresponding parameter.

So we'd never have "actual parameter", and we'd never have "formal argument".

But in this case the thing whose name is `augmented` is the constant variable which is passed as an actual argument to the constructor invocation which is implied by each enum value declaration.

So what we're saying is that the identifier expression `augmented` cannot be used in an enum value declaration. Perhaps something like the following would be more helpful to the reader?:

```suggestion
/// with an augmenting declaration that passes `augmented` as an actual argument.
```

**Review by eernstg — COMMENTED** [review]
I think I understand A06_t04 better now, PTAL.

**Comment by sgrekhov** [comment]
Thank you! Updated. PTAL

**Review by eernstg — APPROVED** [review]
LGTM!

File: LanguageFeatures/Augmentation-libraries/augmented_expression_A06_t04.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (eernstg) [reply LanguageFeatures/Augmentation-libraries/augmented_expression_A06_t04.dart]: Thx!

**Comment by eernstg** [comment]
Landing. Fixed a couple of typos (an `on` clause on an augmenting `extension` declaration, a missing value section in an augmenting `enum` declaration).

Can you run the spec parser locally? It would be good to catch syntax errors a bit earlier.

**Comment by sgrekhov** [comment]
Thank you! Will do with the parser

**Comment by eernstg** [comment]
Thanks!

**Pull request merged**